### PR TITLE
Added signup endpoint for AppUsers with non-guest role to signup.

### PR DESCRIPTION
**Pull request by thaigillespie**

### CODE REVIEW

File: src/controllers/AuthController.ts [hunk 1]
```diff
@@ -1,175 +1,39 @@
...
-    res.status(403).json({
-      msg: ERR_MSG.USER_NOT_WHITELISTED,
-    });
-    return next();
-  }
-
-  const userDocID: string = userDocDetails.uid;
-
-  let newUser: admin.auth.UserRecord;
+    const appUserWhitelisted = await this.appUserService.getAppUserByEmail(email);
```
Comment (godwinpang) [inline src/controllers/AuthController.ts:19]: I think we can drop the whitelist here

File: src/services/FirebaseAdminService.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import admin from 'firebase-admin';
+import '@firebase/auth';
+
```
Comment (godwinpang) [inline src/services/FirebaseAdminService.ts:3]: can we make this a class?

Comment (godwinpang) [reply src/services/FirebaseAdminService.ts:3]: I also think it'd be nice to call this AuthService or AccountService, that way it makes sense if we later don't use firebase anymore

File: src/services/FirebaseAdminService.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+export const createNewFirebaseUser = async (
+  id: number,
+  email: string,
+  password: string
+): Promise<admin.auth.UserRecord | undefined> => {
```
Comment (godwinpang) [inline src/services/FirebaseAdminService.ts:8]: To blackbox this service properly I don't think we should return UserRecord. how about just returning the id of the created user?

File: src/services/AppUserService.ts [hunk 4]
```diff
@@ -119,6 +119,14 @@ export class AppUserService {
 
     return false;
   }
+
+  isGuest(appUser: AppUser): boolean {
+    if (appUser.role !== AppUserRole.GUEST) {
```
Comment (godwinpang) [inline src/services/AppUserService.ts:124]: `return appUser.role === AppUserRole.GUEST`

**Review by godwinpang — CHANGES_REQUESTED** [review]

File: src/services/FirebaseAdminService.ts (hunk 2 above)
Comment (thaigillespie) [reply src/services/FirebaseAdminService.ts:3]: I'll call it SignupService. How does that sound?

File: src/services/FirebaseAdminService.ts (hunk 3 above)
Comment (thaigillespie) [reply src/services/FirebaseAdminService.ts:8]: Yep, sounds good

File: src/services/AppUserService.ts (hunk 4 above)
Comment (thaigillespie) [reply src/services/AppUserService.ts:124]: I keep forgetting about this lol...

File: src/controllers/AuthController.ts (hunk 1 above)
Comment (thaigillespie) [reply src/controllers/AuthController.ts:19]: Renamed to appUserFromEmail

File: src/services/FirebaseAdminService.ts (hunk 2 above)
Comment (godwinpang) [reply src/services/FirebaseAdminService.ts]: I think in general it's better to name things a little more generic - later on we can expose deleteAccount, setVerifyEmail to write scripts and maybe manage accounts. Let's go with accountservice?

Comment (thaigillespie) [reply src/services/FirebaseAdminService.ts]: Yep, changed.

File: src/services/AccountService.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+   * @returns {string} The userID of the newly created Firebase Auth user, if the creation
+   * process succeeds.
+   */
+  async createNewFirebaseUser(
+    id: number,
+    email: string,
+    password: string
+  ): Promise<string | undefined> {
+    try {
+      const newFirebaseUser = await admin.auth().createUser({
```
Comment (godwinpang) [inline src/services/AccountService.ts:20]: later we can inject admin.auth() for testing, but lets merge in for now

**Review by godwinpang — APPROVED** [review]

**Pull request merged**